x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/query/LatLonShapeDocValuesQueryTests.java

```diff
@@ -0,0 +1,169 @@
+                w.addDocument(doc);
+
+
+            } catch (Exception e) {
+                // ignore
+            }
+        }
+
+        if (random().nextBoolean()) {
+            w.forceMerge(1);
+        }
+        final IndexReader r = DirectoryReader.open(w);
+        w.close();
+
+        IndexSearcher s = newSearcher(r);
+        for (int i = 0; i < 100; i++) {
+            org.apache.lucene.geo.Polygon q = GeoTestUtil.nextPolygon();
+            Query indexQuery = LatLonShape.newGeometryQuery(FIELD_NAME, ShapeField.QueryRelation.INTERSECTS, q);
+            Query docValQuery = new LatLonShapeDocValuesQuery(FIELD_NAME,ShapeField.QueryRelation.INTERSECTS, q);
+            assertEquals(s.count(indexQuery), s.count(docValQuery));
+            indexQuery = LatLonShape.newGeometryQuery(FIELD_NAME, ShapeField.QueryRelation.WITHIN, q);
+            docValQuery = new LatLonShapeDocValuesQuery(FIELD_NAME, ShapeField.QueryRelation.WITHIN, q);
+            assertEquals(s.count(indexQuery), s.count(docValQuery));
+            indexQuery = LatLonShape.newGeometryQuery(FIELD_NAME, ShapeField.QueryRelation.DISJOINT, q);
+            docValQuery = new LatLonShapeDocValuesQuery(FIELD_NAME, ShapeField.QueryRelation.DISJOINT, q);
+            assertEquals(s.count(indexQuery), s.count(docValQuery));
+            indexQuery = LatLonShape.newGeometryQuery(FIELD_NAME, ShapeField.QueryRelation.CONTAINS, q);
+            docValQuery = new LatLonShapeDocValuesQuery(FIELD_NAME, ShapeField.QueryRelation.CONTAINS, q);
+            assertEquals(s.count(indexQuery), s.count(docValQuery));
+        }
+        IOUtils.close(r, dir);
+    }
+
+    public void testIndexMultiShapes() throws Exception {
+        IndexWriterConfig iwc = newIndexWriterConfig();
+        // Else seeds may not reproduce:
+        iwc.setMergeScheduler(new SerialMergeScheduler());
+        // Else we can get O(N^2) merging:
+        iwc.setMaxBufferedDocs(10);
+        Directory dir = newDirectory();
+        // RandomIndexWriter is too slow here:
+        IndexWriter w = new IndexWriter(dir, iwc);
+        final int numDocs = randomIntBetween(10, 1000);
+        GeoShapeIndexer indexer = new GeoShapeIndexer(true, FIELD_NAME);
+        for (int id = 0; id < numDocs; id++) {
+            Document doc = new Document();
+            Geometry geometry = GeometryTestUtils.randomGeometryWithoutCircle(randomIntBetween(1, 5), false);
+            try {
+                geometry = indexer.prepareForIndexing(geometry);
+                List<IndexableField> fields = indexer.indexShape(null, geometry);
+                for (IndexableField field : fields) {
+                    doc.add(field);
+                }
+                BinaryGeoShapeDocValuesField docVal = new BinaryGeoShapeDocValuesField(FIELD_NAME);
+                docVal.add(fields, geometry);
+                doc.add(docVal);
+                w.addDocument(doc);
+            } catch (Exception e) {
+                // ignore
```

Comment:
is it safe? Can we narrow down the set of exceptions types we ignore?

Reply:
I remove the try / catch